src/overlays/actors/ovl_Obj_Switch/z_obj_switch.c

```diff
@@ -5,17 +5,79 @@
 
-#if 0
+void ObjSwitch_FloorSwitchUpInit(ObjSwitch* this);
+void ObjSwitch_FloorSwitchPushDownInit(ObjSwitch* this);
+void ObjSwitch_FloorSwitchDownInit(ObjSwitch* this);
+void ObjSwitch_FloorSwitchRiseUpInit(ObjSwitch* this);
+void ObjSwitch_EyeSwitchFrozenInit(ObjSwitch* this);
+void ObjSwitch_EyeSwitchOpenInit(ObjSwitch* this);
+void ObjSwitch_EyeSwitchClosingInit(ObjSwitch* this);
+void ObjSwitch_EyeSwitchClosedInit(ObjSwitch* this);
+void ObjSwitch_EyeSwitchOpeningInit(ObjSwitch* this);
+void ObjSwitch_CrystalSwitchOffInit(ObjSwitch* this);
+void ObjSwitch_CrystalSwitchTurnOnInit(ObjSwitch* this);
+void ObjSwitch_CrystalSwitchOnInit(ObjSwitch* this);
+void ObjSwitch_CrystalSwitchTurnOffInit(ObjSwitch* this);
+void ObjSwitch_LargeFloorSwitchUpInit(ObjSwitch* this);
+void ObjSwitch_LargeFloorSwitchPushDownInit(ObjSwitch* this);
+void ObjSwitch_LargeFloorSwitchDownInit(ObjSwitch* this);
+void ObjSwitch_LargeFloorSwitchRiseUpInit(ObjSwitch* this);
+
+void func_8093AF54(ObjSwitch* this, GlobalContext* globalCtx);
+void ObjSwitch_TryPlayCutscene(ObjSwitch* this, GlobalContext* globalCtx);
+void ObjSwitch_FloorSwitchUp(ObjSwitch* this, GlobalContext* globalCtx);
+void ObjSwitch_FloorSwitchPushDown(ObjSwitch* this, GlobalContext* globalCtx);
+void ObjSwitch_FloorSwitchDown(ObjSwitch* this, GlobalContext* globalCtx);
+void ObjSwitch_FloorSwitchRiseUp(ObjSwitch* this, GlobalContext* globalCtx);
+void ObjSwitch_EyeSwitchUnfrozen(ObjSwitch* this, GlobalContext* globalCtx);
+void ObjSwitch_EyeSwitchOpen(ObjSwitch* this, GlobalContext* globalCtx);
+void ObjSwitch_EyeSwitchClosing(ObjSwitch* this, GlobalContext* globalCtx);
+void ObjSwitch_EyeSwitchClosed(ObjSwitch* this, GlobalContext* globalCtx);
+void ObjSwitch_EyeSwitchOpening(ObjSwitch* this, GlobalContext* globalCtx);
+void ObjSwitch_CrystalSwitchOff(ObjSwitch* this, GlobalContext* globalCtx);
+void ObjSwitch_CrystalSwitchTurnOn(ObjSwitch* this, GlobalContext* globalCtx);
+void ObjSwitch_CrystalSwitchOn(ObjSwitch* this, GlobalContext* globalCtx);
+void ObjSwitch_CrystalSwitchTurnOff(ObjSwitch* this, GlobalContext* globalCtx);
+void ObjSwitch_LargeFloorSwitchUp(ObjSwitch* this, GlobalContext* globalCtx);
+void ObjSwitch_LargeFloorSwitchPushDown(ObjSwitch* this, GlobalContext* globalCtx);
+void ObjSwitch_LargeFloorSwitchDown(ObjSwitch* this, GlobalContext* globalCtx);
+void ObjSwitch_LargeFloorSwitchRiseUp(ObjSwitch* this, GlobalContext* globalCtx);
+
+void ObjSwitch_DrawFloorSwitch(ObjSwitch* this, GlobalContext* globalCtx);
+void ObjSwitch_DrawRustyFloorSwitch(ObjSwitch* this, GlobalContext* globalCtx);
+void ObjSwitch_DrawVisibleEyeSwitch(ObjSwitch* this, GlobalContext* globalCtx);
+void ObjSwitch_DrawInvisibleEyeSwitch(ObjSwitch* this, GlobalContext* globalCtx);
+void ObjSwitch_DrawEyeSwitch(ObjSwitch* this, GlobalContext* globalCtx);
+void ObjSwitch_DrawCrystalSwitch(ObjSwitch* this, GlobalContext* globalCtx);
+
+static void* sEyeSwitchTextures[][4] = {
```

Comment:
```suggestion
static TexturePtr sEyeSwitchTextures[][4] = {
```
We have a typedef specifically for textures.